src/momento/vector_index_client_async.py

```diff
@@ -0,0 +1,182 @@
+from __future__ import annotations
+
+from types import TracebackType
+from typing import Optional, Type
+
+from momento import logs
+from momento.auth import CredentialProvider
+from momento.config import Configuration
+
+try:
+    from momento.internal._utilities import _validate_request_timeout
+    from momento.internal.aio._scs_control_client import _ScsControlClient
+    from momento.internal.aio._vector_index_client import _VectorIndexClient
+except ImportError as e:
+    if e.name == "cygrpc":
+        import sys
+
+        print(
+            "There is an issue on M1 macs between GRPC native packaging and Python wheel tags. "
+            "See https://github.com/grpc/grpc/issues/28387",
+            file=sys.stderr,
+        )
+        print("-".join("" for _ in range(99)), file=sys.stderr)
+        print("    TO WORK AROUND:", file=sys.stderr)
+        print("    * Install Rosetta 2", file=sys.stderr)
+        print(
+            "    * Install Python from python.org (you might need to do this if you're using an arm-only build)",
```

Comment:
is this actually still a thing?

Reply:
I don't know if we need these rosetta workarounds anymore, I feel like this got fixed in gRPC. Can address in a subsequent PR but I would love to get rid of this.

Reply:
I don't know if it's still a thing. I copied it from the data client. Will be good to revisit and can do in a future PR.